pkg/cdi/cdi.go

```diff
@@ -0,0 +1,74 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdi
+
+import (
+	"fmt"
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	cdiSpecs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"github.com/golang/glog"
+	//"sort"
+	//"strings"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+// CreateCDISpec creates CDI spec file with specified devices
+func CreateCDISpec(resourcePrefix string, filteredDevices []types.HostDevice, rPool types.ResourcePool) error {
+	cdiDevices := make([]cdiSpecs.Device, 0)
+	cdiSpec := cdiSpecs.Spec{
+		Version: "0.5.0",
+		Kind:    resourcePrefix + "/net",
+		Devices: cdiDevices,
+	}
+
+	for _, dev := range filteredDevices {
+		containerEdit := cdiSpecs.ContainerEdits{
+			DeviceNodes: make([]*cdiSpecs.DeviceNode, 0),
+			Env:         make([]string, 0),
+		}
+
+		envs, err := rPool.GetEnvs(resourcePrefix, []string{dev.GetDeviceID()})
+		if err != nil {
+			glog.Errorf("CreateCDISpec(): Unable to create environment variables: %s", err)
+			return fmt.Errorf("unable to create environment variables: %s", err)
+		}
+		for name, value := range envs {
+			containerEdit.Env = append(containerEdit.Env, name+"="+value)
+		}
+
+		for i, spec := range dev.GetDeviceSpecs() {
+			glog.Infof("Checking device spec (%d): %s", i, spec.String())
+			deviceNode := cdiSpecs.DeviceNode{
+				Path:        spec.ContainerPath,
+				HostPath:    spec.HostPath,
+				Permissions: "rwm",
```

Comment:
did you check if the m in Permissions is needed?

Reply:
it doesn't needed for VFs. I'll check with SFs and create a separate PR to fix it over the whole project

Reply:
Did you check with and without RDMA CM?

Reply:
yes, I checked both for VFs